build/Makefile

```diff
@@ -127,10 +127,15 @@ install: ensure-build-image
 		agones $(mount_path)/install/helm/agones/
 
 # Build a static binary for the gameserver controller
-build-controller-binary: ensure-build-image
+build-controller-binary: ensure-build-image lint
```

Comment:
I feel like `lint` should run on `make test` ([here](https://github.com/GoogleCloudPlatform/agones/blob/master/build/Makefile#L101)) 

wdyt?

Reply:
Makes sense, removed in build targets and put it in test one.